### PR TITLE
List cmd

### PR DESCRIPTION
**Pull request by amjith**
Adds a `list` command to list all available advices.

0_2_fish_-__fish__Users_amjith_netflix_github_advice-animal_

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
     runner = CliRunner()
     result = runner.invoke(main, ["check", "."])
     assert result.exit_code == 0
+
+
+def test_list():
+    runner = CliRunner()
+    result = runner.invoke(main, ["list"])
+    assert (
+        "click_version_option : GREEN \nruff_show_source : YELLOW (preview)\nuse_py_typed : GREEN \n"
```
Comment (thatch) [inline tests/test_cli.py:27]: Probably more stable would be use the testing advice dir, which currently only has one called "shouty".  Feel free to add ones called "green", "yellow", "red", and "preview" there too, they don't even really need to perform an action.

See existing use at https://github.com/advice-animal/advice-animal/blob/8252255b168f9b5ad782862f63067a987f7c0397/tests/test_runner.py#L10 (it's just override ADVICE_DIR to tests/advice if you're in the cli)

File: advice_animal/cli.py [hunk 2]
```diff
@@ -188,6 +188,19 @@ def check(ctx: click.Context, target: str) -> None:
             )
 
 
+@main.command()
+@click.pass_context
+@click.argument("target", default=".")
```
Comment (thatch) [inline advice_animal/cli.py:193]: Is target used?

**Comment by thatch** [comment]
+1 on this being a thing, a couple minor comments before it's ready to go

File: tests/test_cli.py (hunk 1 above)
Comment (amjith) [reply tests/test_cli.py]: I've updated the tests. I've also changed the output format.

**Comment by amjith** [comment]
This is the new output. 

0_0_fish_-____n_g_advice-animal_

File: advice_animal/cli.py [hunk 3]
```diff
@@ -188,6 +188,20 @@ def check(ctx: click.Context, target: str) -> None:
...
+@click.pass_context
+def list(ctx: click.Context) -> None:
+    runner = Runner(Path(ctx.obj.advice_path), inplace=False, mode=Mode.check)
+    for advice_name, check_cls in runner.iter_check_classes(
+        FixConfidence.UNSET, True, re.compile(".*")
+    ):
+        if check_cls.confidence.name != "UNSET":
+            name = click.style(advice_name, fg=check_cls.confidence.name.lower())
+        else:
+            name = click.style(advice_name, fg="green")
```
Comment (thatch) [inline advice_animal/cli.py:201]: nit: UNSET is the most restrictive, I'd probably go for something "worse" than red, like magenta.

File: advice_animal/cli.py [hunk 4]
```diff
@@ -188,6 +188,20 @@ def check(ctx: click.Context, target: str) -> None:
...
 
+@main.command()
+@click.pass_context
+def list(ctx: click.Context) -> None:
+    runner = Runner(Path(ctx.obj.advice_path), inplace=False, mode=Mode.check)
+    for advice_name, check_cls in runner.iter_check_classes(
+        FixConfidence.UNSET, True, re.compile(".*")
+    ):
+        if check_cls.confidence.name != "UNSET":
+            name = click.style(advice_name, fg=check_cls.confidence.name.lower())
```
Comment (thatch) [inline advice_animal/cli.py:199]: nit: we probably ought to have better color handling now that this conveys an entire piece of information.  Here's how hdeps does it: https://github.com/hdeps/hdeps/blob/087fa55a8b772179b11a21aaa8462c8938a2b6c3/hdeps/cli.py#L152-L155

**Review by thatch — APPROVED** [review]

File: advice_animal/cli.py (hunk 3 above)
Comment (amjith) [reply advice_animal/cli.py:201]: Ah. I assumed UNSET would default to green. My bad.

**Pull request merged**